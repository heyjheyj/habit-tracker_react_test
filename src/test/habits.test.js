import Habits from "../Components/habits";
import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

describe("habits component", () => {
  const habits = [
    {
      name: "Reading",
      count: 4,
      id: 1
    },
    {
      name: "Eating",
      count: 5,
      id: 2
    }
  ];
  let HabitsComponent;
  let onIncrement;
  let onDecrement;
  let onDelete;
  let onAdd;
  let onReset;

  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();
    onAdd = jest.fn();
    onReset = jest.fn();
    HabitsComponent = (
      <Habits
        habits={habits}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
        onAdd={onAdd}
        onReset={onReset}
      />
    );
  });

  it("renders", () => {
    const component = renderer.create(HabitsComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  //   describe("button click", () => {
  //     beforeEach(() => {
  //       render(HabitsComponent);
  //     });

  // it('calls onAdd when Clicking the "Add" button', () => {
  //   const input = screen.getByPlaceholderText("Habit");
  //   const button = screen.getByText("등록");
  //   const newHabit = "New Habit";
  //   userEvent.type(input, newHabit);
  //   userEvent.click(button);
  //   expect(onAdd).toHaveBeenCalledWith(newHabit);
  // });

  // it('calls onIncrement when clicking the "increase" button', () => {
  //   const button = screen.getAllByTitle("increase")[0];
  //   userEvent.click(button);
  //   expect(onIncrement).toHaveBeenCalledWith(habits[0]);
  // });
  //   });
});
