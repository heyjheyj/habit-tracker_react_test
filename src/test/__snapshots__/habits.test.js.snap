// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habits component renders 1`] = `
Array [
  <div>
    <form
      className="inputField"
      onSubmit={[Function]}
    >
      <input
        className="inputText"
        placeholder="취미를 입력해주세요"
        type="text"
      />
      <button
        className="inputButton"
      >
        등록
      </button>
    </form>
  </div>,
  <ul>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Reading
      </span>
      <span
        className="habit-count"
      >
        4
      </span>
      <button
        className="habit-button habit-increase"
        onClick={[Function]}
        title="increase"
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        onClick={[Function]}
        title="decrease"
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        onClick={[Function]}
        title="delete"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Eating
      </span>
      <span
        className="habit-count"
      >
        5
      </span>
      <button
        className="habit-button habit-increase"
        onClick={[Function]}
        title="increase"
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        onClick={[Function]}
        title="decrease"
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        onClick={[Function]}
        title="delete"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>,
  <section
    className="reset"
  >
    <button
      className="resetButton"
      onClick={[MockFunction]}
    >
      Reset All
    </button>
  </section>,
]
`;
